From 7f4415d01a245e1f97e6b6ee849db36ba4d3b15c Mon Sep 17 00:00:00 2001
From: Aharon Campos <aharon.campos@continental-corporation.com>
Date: Sat, 19 Nov 2011 20:09:12 -0600
Subject: [PATCH 1/3] Cap xdelta memory consumption on target while patching

(From CCase patch 001_cap_xdelta_memory_consumption.patch
submitted on Aug 24, 2011)
---
 xdelta3.h |   12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/xdelta3.h b/xdelta3.h
index 5dafd8d..08f1ad3 100644
--- a/xdelta3.h
+++ b/xdelta3.h
@@ -38,19 +38,19 @@
  * buffer is used directly.
  */
 #ifndef XD3_DEFAULT_WINSIZE
-#define XD3_DEFAULT_WINSIZE (1U << 23)
+#define XD3_DEFAULT_WINSIZE (1U << 20)
 #endif
 
 /* Default total size of the source window used in xdelta3-main.h */
 #ifndef XD3_DEFAULT_SRCWINSZ
-#define XD3_DEFAULT_SRCWINSZ (1U << 26)
+#define XD3_DEFAULT_SRCWINSZ (1U << 23)
 #endif
 
 /* When Xdelta requests a memory allocation for certain buffers, it
  * rounds up to units of at least this size.  The code assumes (and
  * asserts) that this is a power-of-two. */
 #ifndef XD3_ALLOCSIZE
-#define XD3_ALLOCSIZE (1U<<14)
+#define XD3_ALLOCSIZE (1U<<11)
 #endif
 
 /* The XD3_HARDMAXWINSIZE parameter is a safety mechanism to protect
@@ -60,7 +60,7 @@
  *
  * 8-16MB is reasonable, probably don't need to go larger. */
 #ifndef XD3_HARDMAXWINSIZE
-#define XD3_HARDMAXWINSIZE (1U<<24)
+#define XD3_HARDMAXWINSIZE (1U<<21)
 #endif
 /* The IOPT_SIZE value sets the size of a buffer used to batch
  * overlapping copy instructions before they are optimized by picking
@@ -68,12 +68,12 @@
  * longer a forced xd3_srcwin_setup() decision is held off.  Setting
  * this value to 0 causes an unlimited buffer to be used. */
 #ifndef XD3_DEFAULT_IOPT_SIZE
-#define XD3_DEFAULT_IOPT_SIZE    (1U<<15)
+#define XD3_DEFAULT_IOPT_SIZE    (1U<<12)
 #endif
 
 /* The maximum distance backward to search for small matches */
 #ifndef XD3_DEFAULT_SPREVSZ
-#define XD3_DEFAULT_SPREVSZ (1U<<18)
+#define XD3_DEFAULT_SPREVSZ (1U<<15)
 #endif
 
 /* The default compression level
-- 
1.7.9.5

